Build failure: sunshine with Cuda Support

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Updated my flake lock and immediately noticed the cuda support for sunshine is completely broken, the build fails immediately saying "Cuda not found"

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286318161

### Relevant log output

```console
Sourcing auto-add-driver-runpath-hook
Using autoAddDriverRunpath
Sourcing fix-elf-files.sh
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/6kiim52627p7mifnr3h20ax1rjygzlcy-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/share/doc/Sunshine -DCMAKE_INSTALL_INFODIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/include -DCMAKE_INSTALL_SBINDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/ymildcka4gwnhs34lkxbjdrpdx98dwns-binutils-wrapper-2.43.1/bin/strip -DCMAKE_RANLIB=/nix/store/ymildcka4gwnhs34lkxbjdrpdx98dwns-binutils-wrapper-2.43.1/bin/ranlib -DCMAKE_AR=/nix/store/ymildcka4gwnhs34lkxbjdrpdx98dwns-binutils-wrapper-2.43.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/mhzvklabm9daqk3nwskrq7v9dr92z0h9-sunshine-2025.118.151840 -Wno-dev -DUDEV_FOUND:BOOL=TRUE -DSYSTEMD_FOUND:BOOL=TRUE -DUDEV_RULES_INSTALL_DIR:STRING=lib/udev/rules.d -DSYSTEMD_USER_UNIT_INSTALL_DIR:STRING=lib/systemd/user -DBOOST_USE_STATIC:BOOL=FALSE -DBUILD_DOCS:BOOL=FALSE -DSUNSHINE_PUBLISHER_NAME:STRING=nixpkgs -DSUNSHINE_PUBLISHER_WEBSITE:STRING=https://nixos.org -DSUNSHINE_PUBLISHER_ISSUE_URL:STRING=https://github.com/NixOS/nixpkgs/issues
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/5xxsgsypi8a31glic5p8wmqx6ilipiw6-gcc-wrapper-13.3.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/5xxsgsypi8a31glic5p8wmqx6ilipiw6-gcc-wrapper-13.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
[0mGot from CI master branch and version 2025.118.151840[0m
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Boost include dirs: /nix/store/jkwsbly2wga09y64hnx5sckql6vwqhsj-boost-1.87.0-dev/include
-- Boost libraries: Boost::filesystem;Boost::locale;Boost::log;Boost::program_options;Boost::system
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for poll
-- Looking for poll - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for gethostbyaddr_r
-- Looking for gethostbyaddr_r - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Performing Test HAS_MSGHDR_FLAGS
-- Performing Test HAS_MSGHDR_FLAGS - Success
-- Check size of socklen_t
-- Check size of socklen_t - done
-- Check size of QOS_FLOWID
-- Check size of QOS_FLOWID - failed
-- Check size of PQOS_FLOWID
-- Check size of PQOS_FLOWID - failed
-- Found Boost: /nix/store/jkwsbly2wga09y64hnx5sckql6vwqhsj-boost-1.87.0-dev/lib/cmake/Boost-1.87.0/BoostConfig.cmake (found suitable version "1.87.0", minimum required is "1.53.0") found components: system
-- Found OpenSSL: /nix/store/5926bc2bqkr0882p4m6yq3wy5brnbwlv-openssl-3.3.2/lib/libcrypto.so (found version "3.3.2")
[33mCMake Warning at third-party/libdisplaydevice/src/CMakeLists.txt:14 (message):
  Linux is not supported yet.

[0m
-- Found PkgConfig: /nix/store/q5h7fl6q9rza6q300nnpmf5fgg6c73a6-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libcurl'
--   Found libcurl, version 8.11.1
-- Checking for module 'miniupnpc'
--   Found miniupnpc, version 2.2.8
-- Looking for a CUDA compiler
-- Looking for a CUDA compiler - NOTFOUND
[31mCMake Error at cmake/compile_definitions/linux.cmake:71 (message):
  CUDA not found.

                  If this is intentional, set '-DSUNSHINE_ENABLE_CUDA=OFF' or '-DCUDA_FAIL_ON_MISSING=OFF'
Call Stack (most recent call first):
  cmake/compile_definitions/common.cmake:47 (include)
  CMakeLists.txt:57 (include)

[0m
-- Configuring incomplete, errors occurred!
```

### Additional context

Updated flake.lock
```
• Updated input 'nixpkgs':
    'github:nixos/nixpkgs/bffc22eb12172e6db3c5dde9e3e5628f8e3e7912?narHash=sha256-8YVQ9ZbSfuUk2bUf2KRj60NRraLPKPS0Q4QFTbc%2Bc2c%3D' (2025-01-08)
  → 'github:nixos/nixpkgs/9e4d5190a9482a1fb9d18adf0bdb83c6e506eaab?narHash=sha256-nmKOgAU48S41dTPIXAq0AHZSehWUn6ZPrUKijHAMmIk%3D' (2025-01-21)
```
I'm using kernel `linuxPackages_6_12` and I have cuda enabled globally via `nixpkgs.config.cudaSupport = lib.mkForce true;`

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.8, NixOS, 25.05 (Warbler), 25.05.20250108.bffc22e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.25.3`
 - nixpkgs: `/nix/store/v0g0bxsd5gw6k0jz2855f8h7l1218925-source`

### Notify maintainers

@devusb 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/su/sunshine/package.nix
+++ b/pkgs/by-name/su/sunshine/package.nix
@@ -140,6 +140,7 @@ stdenv'.mkDerivation rec {
     ]
     ++ lib.optionals cudaSupport [
       cudaPackages.cudatoolkit
+      cudaPackages.cuda_cudart
     ]
     ++ lib.optionals stdenv.hostPlatform.isx86_64 [
       intel-media-sdk

